### ISSUE TITLE
Implicit conversion between bool and QuadOption

### ISSUE THREAD

**Issue by fekir** [neomutt/neomutt#1765]
Hello, I have a question about the correctness/understandability about this (init.c:3343:39, function `enum QuadOption query_quadoption(enum QuadOption opt, const char *prompt)`) line of code:

````
opt = mutt_yesorno(prompt, (opt == MUTT_ASKYES));
````

I assumed `mutt_yesorno` would take as second parameter a `bool`, but it actually takes an `enum QuadOption`.

In the `mutt_yesorno` function, the parameter is evaluated here:

````
mutt_str_asprintf(&answer_string, " ([%s]/%s): ", (def == MUTT_YES) ? yes : no,
                    (def == MUTT_YES) ? no : yes);
````
Thus given

````
enum QuadOption
{
  MUTT_ABORT = -1, ///< User aborted the question (with Ctrl-G)
  MUTT_NO,         ///< User answered 'No', or assume 'No'
  MUTT_YES,        ///< User answered 'Yes', or assume 'Yes'
  MUTT_ASKNO,      ///< Ask the user, defaulting to 'No'
  MUTT_ASKYES,     ///< Ask the user, defaulting to 'Yes'
};
````

if `opt == MUTT_ASKYES` -> converts to `1`, converts to `MUTT_YES` -> prints `yes`
if `opt != MUTT_ASKYES` -> converts to `0`, converts to `MUTT_NO` -> prints `no`

`MUTT_YES` and `MUTT_NO` are handled separately, thus `opt` could be `MUTT_ABORT` too.
I'm unable to see if this scenario is possible,but would it be the correct behaviour?

If not, then a switch case is missing.

Otherwise I would propose to change the code to 

````
opt = mutt_yesorno(prompt, (opt == MUTT_ASKYES) ? MUTT_YES : MUTT_NO);
````

In order not to rely on the specific `enum` values.

The same question holds for `mutt_yesorno`, when called with something different than `MUTT_YES` and `MUTT_NO`.

The documentation states that `* @param def Default answer, see #QuadOption`, but in reality this is true if `def` is only `MUTT_YES` or `MUTT_NO`.
Is it thus a programming error calling the function with something else or the documentation that does not reflect the intended functionality?

**Comment by flatcap** [comment]
Thanks for the detailed analysis.
At times like these, it's good to check the upstream Mutt source.
Yep, there's a bug.  They rely on the fact that a boolean `true` is equivalent to `MUTT_YES`.

> `opt = mutt_yesorno(prompt, (opt == MUTT_ASKYES) ? MUTT_YES : MUTT_NO);`

Yes, that'll do.

> @param def Default answer, see #QuadOption

Yes, this would be better: `@param def Default answer, #MUTT_YES or #MUTT_NO`

**Comment by flatcap** [comment]
Fixed by 8df8f8ccc
